src/decompiler-naming.h

```diff
@@ -110,6 +110,40 @@ void RenameToIdentifiers(std::vector<T*>& things, BindingHash& bh,
   }
 }
 
+enum {
+  min_content_identifier_size = 7,
+  max_content_identifier_size = 30
+};
+
+void RenameToContents(std::vector<DataSegment*>& segs, BindingHash& bh) {
+  std::string s;
+  for (auto seg : segs) {
+    s = "d_";
+    for (auto c : seg->data) {
+      if (isalnum(c) || c == '_') {
+        s += static_cast<char>(c);
+      }
+      if (s.size() >= max_content_identifier_size) {
+        break;
+      }
+    }
+    if (s.size() < min_content_identifier_size) {
```

Comment:
My first thought was "why have a minimum at all?" Probably worth a comment saying why too-short names aren't useful (which is alluded to in the following comment about meaningful binary data)

Reply:
will do